src/test/scala/org/constellation/infrastructure/redownload/RedownloadPeriodicCheckTest.scala

```diff
@@ -21,9 +21,9 @@ class RedownloadPeriodicCheckTest
   before {
     dao = TestHelpers.prepareMockedDAO()
 
-    dao.redownloadService.fetchPeersProposals() shouldReturnF Unit
-    dao.redownloadService.recalculateMajoritySnapshot() shouldReturnF Unit
-    dao.redownloadService.checkForAlignmentWithMajoritySnapshot() shouldReturnF Unit
+    dao.redownloadService.fetchAndSetPeersProposals() shouldReturnF Map()
+    dao.redownloadService.recalculateMajoritySnapshot() shouldReturnF (Seq[RecentSnapshot](), Set[Id]())
```

Comment:
Haven't we talked on slack that `recalculateMajoritySnapshot` should be called internally from `checkForAlignmentWithMajoritySnapshot` on demand?

Reply:
It does, you want to make it private? How would this be tested?

Reply:
So if there is a huge block of code to test it means you're doing something wrong or you're not testing it as a unit but with all dependencies. In fact, you should i.e. mock majority state chooser, API, etc. and check only the redownload service functionality to call particular methods (public interface of other parts of the system, services i.e.) or the behavior

Reply:
Also, only the public interface should be tested. The internal stuff should never be exposed publicly and tested. It leads to very simple conclusion - if unit tests covers all the cases, all private code is tested as well. Otherwise, it's a "code smell" so wrongly written code or too many "unnecessary" code.